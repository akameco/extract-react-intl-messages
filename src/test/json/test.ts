import fs from 'fs'
import path from 'path'
import tempy from 'tempy'
import m from '../..'

test('export json', async () => {
  const tmp = tempy.directory()
  await m(['en', 'ja'], 'src/test/fixtures/default/**/*.js', tmp)
  const en = JSON.parse(fs.readFileSync(path.resolve(tmp, 'en.json'), 'utf8'))
  const ja = JSON.parse(fs.readFileSync(path.resolve(tmp, 'ja.json'), 'utf8'))
  expect(en).toMatchSnapshot()
  expect(ja).toMatchSnapshot()
})

test('export json with removed messages', async () => {
  const tmp = tempy.directory()
  await m(['en', 'ja'], 'src/test/fixtures/default/**/*.js', tmp)
  const enBefore = JSON.parse(
    fs.readFileSync(path.resolve(tmp, 'en.json'), 'utf8')
  )
  const jaBefore = JSON.parse(
    fs.readFileSync(path.resolve(tmp, 'ja.json'), 'utf8')
  )
  expect(enBefore).toMatchSnapshot()
  expect(jaBefore).toMatchSnapshot()
  await m(['en', 'ja'], 'src/test/fixtures/removed/**/*.js', tmp)
  const en = JSON.parse(fs.readFileSync(path.resolve(tmp, 'en.json'), 'utf8'))
  const ja = JSON.parse(fs.readFileSync(path.resolve(tmp, 'ja.json'), 'utf8'))
  expect(en).toMatchSnapshot()
  expect(ja).toMatchSnapshot()
})

test('export json - nest', async () => {
  const tmp = tempy.directory()
  await m(['en', 'ja'], 'src/test/fixtures/default/**/*.js', tmp, {
    defaultLocale: 'en',
    flat: false
  })
  const en = JSON.parse(fs.readFileSync(path.resolve(tmp, 'en.json'), 'utf8'))
  const ja = JSON.parse(fs.readFileSync(path.resolve(tmp, 'ja.json'), 'utf8'))
  expect(en).toMatchSnapshot()
  expect(ja).toMatchSnapshot()
})

test('sort keys', async () => {
  const tmp = tempy.directory()
  const enPath = path.resolve(tmp, 'en.json')
  const jaPath = path.resolve(tmp, 'ja.json')

  await m(['en', 'ja'], 'src/test/fixtures/unsorted/**/*.js', tmp)
  const en = JSON.parse(fs.readFileSync(enPath, 'utf8'))
  const ja = JSON.parse(fs.readFileSync(jaPath, 'utf8'))

  expect(Object.keys(en)).toMatchSnapshot()
  expect(Object.keys(ja)).toMatchSnapshot()
})

test('export using custom module', async () => {
  const tmp = tempy.directory()
  await m(['en', 'ja'], 'src/test/fixtures/custom/**/*.js', tmp, {
    defaultLocale: 'en',
    moduleSourceName: '../i18n'
  })
  const en = JSON.parse(fs.readFileSync(path.resolve(tmp, 'en.json'), 'utf8'))
  const ja = JSON.parse(fs.readFileSync(path.resolve(tmp, 'ja.json'), 'utf8'))
  expect(en).toMatchSnapshot()
  expect(ja).toMatchSnapshot()
})
