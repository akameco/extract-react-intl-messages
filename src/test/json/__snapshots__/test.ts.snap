// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export json - nest 1`] = `
Object {
  "a": Object {
    "hello": "hello",
    "world": "world",
  },
  "b": Object {
    "hello": "hello",
    "world": "world",
  },
}
`;

exports[`export json - nest 2`] = `
Object {
  "a": Object {
    "hello": "",
    "world": "",
  },
  "b": Object {
    "hello": "",
    "world": "",
  },
}
`;

exports[`export json 1`] = `
Object {
  "a.hello": "hello",
  "a.world": "world",
  "b.hello": "hello",
  "b.world": "world",
}
`;

exports[`export json 2`] = `
Object {
  "a.hello": "",
  "a.world": "",
  "b.hello": "",
  "b.world": "",
}
`;

exports[`export json with removed messages 1`] = `
Object {
  "a.hello": "hello",
  "a.world": "world",
  "b.hello": "hello",
  "b.world": "world",
}
`;

exports[`export json with removed messages 2`] = `
Object {
  "a.hello": "",
  "a.world": "",
  "b.hello": "",
  "b.world": "",
}
`;

exports[`export json with removed messages 3`] = `
Object {
  "a.hello": "hello",
  "b.hello": "hello",
  "b.world": "world",
}
`;

exports[`export json with removed messages 4`] = `
Object {
  "a.hello": "",
  "b.hello": "",
  "b.world": "",
}
`;

exports[`export using custom module 1`] = `
Object {
  "a.custom.hello": "hello",
  "a.custom.world": "world",
  "b.custom.message": "Message",
}
`;

exports[`export using custom module 2`] = `
Object {
  "a.custom.hello": "",
  "a.custom.world": "",
  "b.custom.message": "",
}
`;

exports[`sort keys 1`] = `
Array [
  "a.hello",
  "a.world",
  "b.hello",
  "b.world",
  "c.hello",
  "y.hello",
  "z.hello",
]
`;

exports[`sort keys 2`] = `
Array [
  "a.hello",
  "a.world",
  "b.hello",
  "b.world",
  "c.hello",
  "y.hello",
  "z.hello",
]
`;
